### PR TITLE
Romain/refactor

### PR DESCRIPTION
**Pull request by yonromai**
**MERGE THIS PR FIRST: https://github.com/spotify/spotify-tensorflow/pull/14**

### CODE REVIEW

**Comment by yonromai** [comment]
@ravwojdyla PTAL

File: spotify_tensorflow/dataset.py [hunk 1]
```diff
@@ -20,74 +20,61 @@
...
         Args:
             feature_desc_path: filepath to a feature description file.
-            feature_mapping_fn: A function which maps feature spec line to `FixedLenFeature` or
-                `VarLenFeature` values.
+            dir_path: Directory path containing features.
 
         Returns:
-            A Tuple of two elements: (number of features, parse function). Parse function takes a
-            single element - a scalar string Tensor, a single serialized Example.
+            An array containing the names of the features.
```
Comment (ravwojdyla) [inline spotify_tensorflow/dataset.py]: `A list`?

**Comment by yonromai** [comment]
@ravwojdyla this pieces of logic were not test before, I've just moved code around but not added any. I agree that this needs more testing esp. integration / end2end testing. However it doesn't make sense to do it now. We should wait to add support to TFRecordSpec for that. Thoughts?

**Comment by ravwojdyla** [comment]
@yonromai well, it seems like `get_featran_example_dataset` got hidden in `_ get_featran_example_dataset`, and the entry points are now the iterators, thus it makes more sense to tests those, right?

**Comment by yonromai** [comment]
@ravwojdyla I agree. the issue is that the public methods (i.e `get_featran_example_dataset`) take a dataset path as an input. So a natural way to test them is to package some tfrecords test resources (and their corresponding schema) and do some integration testing. 
I looked into it but since the feature schema has changed in scio, I think these tests should be part of the next PR (which will include the updated feature schema): I'd like to avoid writing code to test behaviors that are already outdated and will be replaced in the next PR.  
Thoughts?

**Comment by yonromai** [comment]
@ravwojdyla can we merge? I'll submit another PR tomorrow with the new TFRecordSpec and the integration test.

**Pull request merged**